---
title: "Tuto PhyloseqExt"
output: github_document
---

```{r}
fnFs<-sort(list.files(path, pattern="_R1_001.fastq", full.names=TRUE))
fnRs<-sort(list.files(path, pattern="_R2_001.fastq", full.names=TRUE))
sample.names<-sapply(strsplit(basename(fnFs),"_"),`[`, 1)
```

```{r}
filtFs<-file.path(path,"filtered",paste0(sample.names,"_F_filt.fastq.gz"))
filtRs<-file.path(path,"filtered",paste0(sample.names,"_R_filt.fastq.gz"))

out<-filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(240,160),maxN=0,maxEE=c(2,2),truncQ=2,rm.phix=TRUE,compress=TRUE,multithread=TRUE)

head(out)
```

```{r}
errF<-learnErrors(filtFs, multithread=TRUE)
errR<-learnErrors(filtRs, multithread=TRUE)
```

```{r}
dadaFs<-dada(filtFs, err=errF, multithread=TRUE)
dadaRs<-dada(filtRs, err=errR, multithread=TRUE)
```

```{r}
mergers<-mergePairs(dadaFs,filtFs,dadaRs,filtRs, verbose=TRUE)
```

```{r}
seqtab<-makeSequenceTable(mergers)
```

```{r}
seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=TRUE, verbose=TRUE)
```

```{r}
getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(mergers, getN), rowSums(seqtab.nochim))
colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
rownames(track) <- sample.names
```

```{r}
taxa <- assignTaxonomy(seqtab.nochim, "~/DADA2/silva_nr_v132_train_set.fa.gz", multithread=TRUE)
taxa.print<-taxa
rownames(taxa.print)<-NULL
```

```{r}
unqs.mock<-seqtab.nochim["Mock_F_filt.fastq.gz",]

unqs.mock <- sort(unqs.mock[unqs.mock>0], decreasing=TRUE)

cat("DADA2 inferred", length(unqs.mock), "sample sequences present in the Mock community.\n")
```

```{r}
mock.ref<-getSequences(file.path(path, "HMP_MOCK.v35.fasta"))

match.ref<-sum(sapply(names(unqs.mock), function(x) any(grepl(x, mock.ref))))

cat("Of those,", sum(match.ref), "were exact matches to the expected reference sequences.\n")
```

```{r}
samples.out <- rownames(seqtab.nochim)

subject <- sapply(strsplit(samples.out, "D"), `[`, 1)

gender <- substr(subject,1,1)

subject <- substr(subject,2,999)

day <- as.integer(sapply(strsplit(sample.names, "D"), `[`, 2))

samdf <- data.frame(Subject=subject, Gender=gender, Day=day)

samdf$When <- "Early"

samdf$When[samdf$Day>100] <- "Late"

rownames(samdf) <- samples.out
```

```{r}
ps<-phyloseq(otu_table(seqtab.nochim, taxa_are_rows=FALSE), 
               sample_data(samdf), 
               tax_table(taxa))

ps<-prune_samples(sample_names(ps) !="Mock_F_filt.fastq.gz",ps)
```

```{r}
dna<-Biostrings::DNAStringSet(taxa_names(ps))

names(dna) <- taxa_names(ps)
 
ps <- merge_phyloseq(ps, dna)

taxa_names(ps) <- paste0("ASV", seq(ntaxa(ps)))
```

# IntÃ©grer l'object Phyloseq

```{r}
ps_connect <-url("https://raw.githubusercontent.com/spholmes/F1000_workflow/master/data/ps.rds")
ps = readRDS(ps_connect)
ps
```
```{r}
rank_names(ps)
```

```{r}
table(tax_table(ps)[, "Phylum"], exclude = NULL)
```



